### PR TITLE
Make Date Time Wording More Uniform

### PR DESCRIPTION
**Pull request by cmhedrickREI**
Ticket: GSA/data.gov#5806

- Cleans up wording
    - **"Dataset Last Modified"** = When a dataset was modified by publisher/owners (comes from DCAT `modified` field)
    - **"Dataset Issued"** = When a dataset was issued by publisher/owners (comes from DCAT `issued` field)
    - **"Metadata Last Checked"** = When Harvester last harvested the metadata
- Adds filters to help take different time strings and translates them to datetime objects that can be more easily formated
- Format function for date/datetime objects to help with making formats more uniform

**_Changes live on dev_**

### CODE REVIEW

**Comment by rshewitt** [comment]
here's a recommendation I got from AI as a comparison just to jog some ideas. i'm also cool with your definitions too

```
Dataset Last Updated

  The date provided by the publisher indicating when the dataset or its metadata was last changed.

Dataset First Published

  The date provided by the publisher indicating when the dataset was first made available.

Catalog Last Checked

  The date this catalog last harvested or checked the publisher’s metadata record.
```

File: tests/unit/test_dataset_detail.py [hunk 1]
```diff
@@ -99,9 +99,9 @@ def test_dataset_detail_by_slug(self, interface_with_dataset, db_client):
             for item in dataset_info_box.select(".sidebar-section__item")
         }
         assert "Dataset Issued" in dataset_info_items
-        assert dataset_info_items["Dataset Issued"] == "2021-03-15"
+        assert dataset_info_items["Dataset Issued"] == "March 15, 2021"
         assert "Dataset Last Modified" in dataset_info_items
-        assert dataset_info_items["Dataset Last Modified"] == "2023-06-01"
+        assert dataset_info_items["Dataset Last Modified"] == "June 01, 2023"
```
Comment (rshewitt) [inline tests/unit/test_dataset_detail.py]: do you mind adding a check for "Metadata Last Checked" since we support that one too?

File: tests/unit/test_filters.py [hunk 2]
```diff
@@ -32,3 +41,40 @@ def test_empty_string_returns_none(self):
...
+
+    def test_date_only_string_returns_date(self):
+        result = parse_datetime("2026-05-01")
+        assert type(result) is date
+        assert result == date(2026, 5, 1)
+
+    @pytest.mark.parametrize(
+        "value",
+        [
+            "2026-05-01T00:00:00",
```
Comment (rshewitt) [inline tests/unit/test_filters.py:56]: since they're valid dcatus the datetimes for issued and modified fit a standardized regex format but i wonder if there's anything it would support which fails to parse via `parse_datetime`. seems like an edge case. no need to do it just a thought.

File: tests/unit/test_dataset_detail.py (hunk 1 above)
Comment (cmhedrickREI) [reply tests/unit/test_dataset_detail.py]: covered by line 121 in the test

**Review by rshewitt — APPROVED** [review]
looks good thanks for making those changes!
